Private Events - Complete Projects

GabrielJSuarez opened the pull request:
- Create pull request for the Private Events project
- Created basic authentication with sessions
- Created models for users/events/attendances
- Created queries to divide past and upcoming events
- Added validations in views and databases

GabrielJSuarez committed:
feature:
- Create users controller with new, create and show
- Create Restfull routes for users

GabrielJSuarez committed:
Test:
- Create test with Rspec/Capybara for DB validation, user creation and showing a created user

GabrielJSuarez committed:
Test:
- Create helper methods for alert error messages
- Create helper methods for user login/logout and user sign in

GabrielJSuarez committed:
Test:
- Create events table and added assosiations with User table

GabrielJSuarez committed:
Feature:
- Add association between Users and Events (who creates the event)
Test:
- Add test for associations

GabrielJSuarez committed:
Merge remote-tracking branch 'origin/events-feature' into events-feature

# Conflicts:
#	db/schema.rb

GabrielJSuarez committed:
Feature:
- Add association between Users and Events (who creates the event)
Test:
- Add test for associations

GabrielJSuarez committed:
Feature:
- Add association between Users and Events (who creates the event)
Test:
- Add test for associations

GabrielJSuarez committed:
Merge remote-tracking branch 'origin/events-feature' into events-feature

GabrielJSuarez committed:
Feature:

- Add root page

GabrielJSuarez committed:
Feature:

- Add index/show/new events controller/view/helpers to project

GabrielJSuarez committed:
Feature:

- Create navbar helper to display the user's options when he's logged in

GabrielJSuarez committed:
Feature:

- Add associations to track the attended events and the users who are attending them

Test
- Add rspec association tests for user, events & attendance

GabrielJSuarez committed:
Feature
- Add Attendance controller/view/routes to use the association between user and events

GabrielJSuarez committed:
Feature
- Add style to landing page and it's helper method

GabrielJSuarez committed:
Feature
- Modify helper for events, add event scopes

GabrielJSuarez committed:
Feature
- Add validations for attendances at model level
- Add helper methods to better render views
- Add validation in helper methods to avoid repeated assistances
- Modify styles to better represent past and upcoming events

GabrielJSuarez committed:
Feature
- Change the way that errors are being displayed
- Change the way that the attendance confirmation is being displayed with errors

GabrielJSuarez committed:
Feature
- Change the way that errors are being displayed
- add helper method to error displaying

GabrielJSuarez committed:
Feature
- Add display for users upcoming and past events as a host & attendee

GabrielJSuarez committed:
Test

- Add validation/association tests for users, events, attendances
- Add Integration test for creating users, loging into the page, creating new events

GabrielJSuarez committed:
Rubocop
- Fixed rubocop offenses

GabrielJSuarez committed:
Rubocop
- Fixed rubocop offenses

GabrielJSuarez committed:
Rubocop
- Fixed rubocop offenses

GabrielJSuarez committed:
Update README file

blackpintz commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,23 @@
+<div class="col-md-6 mx-auto text-center mt-3">
+
+  <%= show_single_event(@event) %>
+
+  <% if current_user != @event.creator %>
+    <%= assist_event?(@event) %>
+  <% end %>
+
+  <%= link_to 'View all users events', user_path(@event.creator), class: 'btn btn-primary my-5' %>
+
+  <%= link_to 'View All Events', events_path, class: 'btn btn-warning my-5' %>
+
+  <ul class="list-group">
+    <li class="list-group-item font-weight-bold">
+      Attendance List!
+    </li>
+    <% @event.attendees.each do |person| %>
+      <li class="list-group-item">
+        <%= person.name %>
+      </li>
+    <% end %>
+  </ul>
+</div>
```
There should be no logic (conditional statements and loops) in this file. Create them in View helpers `(app/helpers/*_helper.rb)` instead.

blackpintz commented on app/models/attendance.rb:
```diff
@@ -0,0 +1,6 @@
+class Attendance < ApplicationRecord
+  belongs_to :attendee, class_name: 'User'
+  belongs_to :attended_event, class_name: 'Event'
+
+  validates :attendee_id, uniqueness: { scope: :attended_event_id, message: 'User is already attending this event' }
+end
```
Add spec tests for the above associations and validations.

blackpintz reviewed (changes requested):
# Status: Changes Required

### Hello team 😃,

**Great job on your project 💪. There are, however, a few fixes needed before the project can be approved.**

1). **A logged-in user should be able to attend an event whether it is a future event or past event.**

2). **Add a short description next to the project title in your readme, of what the project is about.**

3). **Add the following files/folders in `.gitignore` and remove them from your git repo: `.idea` and `.browserslistrc`.**

4). **See two more comments below.**

[Optional]

- You have misspelt sign up on your homepage.

- There is code that is displaying at the bottom of every event.

## Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

GabrielJSuarez committed:
- Add .idea folder to .gitignore file
- Move logic to event helper in event show page
- Fixed a typo on homepage
- Add short description to README file about the project, changed yarn add command
- Add attandance validation tests

voscarmv reviewed (approved):
# Project status: Approved ✨ 😎👍 💯 ✔️

- Good work applying all necessary changes.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️
